File: modules/wem-api/src/main/java/com/enonic/wem/api/content/site/SiteTemplateKey.java
package com.enonic.wem.api.content.site;

import static com.google.common.base.Preconditions.checkNotNull;


public final class SiteTemplateKey
{
    private final SiteTemplateName name;

    private SiteTemplateKey( final SiteTemplateName name )
    {
        checkNotNull( name );
        this.name = name;
    }

    public SiteTemplateName getName()
    {
        return name;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || !( o instanceof SiteTemplateKey ) )
        {
            return false;
        }
        final SiteTemplateKey that = (SiteTemplateKey) o;
        return name.equals( that.name );
    }

    @Override
    public int hashCode()
    {
        return name.hashCode();
    }

    @Override
    public String toString()
    {
        return name.toString();
    }

    public static SiteTemplateKey from( final String siteTemplateName )
    {
        return new SiteTemplateKey( new SiteTemplateName( siteTemplateName ) );
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/content/site/template/SiteTemplateKey.ts
module api.content.site.template {

    export class SiteTemplateKey implements api.Equitable {

        private name: string;

        public static fromString(name: string): SiteTemplateKey {
            return new SiteTemplateKey(name);
        }

        constructor(siteTemplateName: string) {
            this.name = siteTemplateName;
        }

        getName(): string {
            return this.name;
        }

        toString(): string {
            return this.name;
        }

        equals(o: api.Equitable): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, SiteTemplateKey)) {
                return false;
            }

            var other = <SiteTemplateKey>o;

            if (!api.ObjectHelper.stringEquals(this.name, other.name)) {
                return false;
            }

            return true;
        }

    }
}
